scripts/Dockerfile.testing

```diff
@@ -0,0 +1,56 @@
+# This is a Dockerfile for one-off builds of stellar-core during development.
+#
+# It's intended to be run from the stellar-core source directory of a
+# developer's workstation to make a test buid that can be run in kubernetes
+# without passing through any of the normal CI process. As such it is as small
+# and quick as possible, using ubuntu linux and neither building nor installing
+# an intermediate debian package.
+#
+# This file does not even require the workspace to be pushed to github, much
+# less in an open PR, only committed to the local git repo so 'git clean' knows
+# which files to keep: it copies the contents of the directory it's invoked from
+# and then does 'git clean' and a rebuild in the container's build environment.
+# There's no audit trail in the resulting images and they should only ever go in
+# user repositories for testing purposes.
+#
+# To use this file, run something like the following:
+#
+#  $ export TAG=${USER}/stellar-core:$(git describe --always --tags --long)
+#  $ docker build -f scripts/Dockerfile.testing . -t $TAG
+#
+
+# Build stage, includes everything needed to build.
+FROM ubuntu:focal as buildstage
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && \
+    apt-get -y install iproute2 procps lsb-release \
+                       git build-essential pkg-config autoconf automake libtool \
+                       bison flex libpq-dev parallel libunwind-dev \
+                       clang-8 libc++abi-8-dev libc++-8-dev \
+                       postgresql
+
+COPY . stellar-core/
+WORKDIR stellar-core
+RUN git clean -dxf
+RUN git submodule foreach --recursive git clean -dxf
+
+ARG CC=clang-8
+ARG CXX=clang++-8
+ARG CFLAGS='-O3 -g1 -fno-omit-frame-pointer'
+ARG CXXFLAGS='-O3 -g1 -fno-omit-frame-pointer -stdlib=libc++'
+
+RUN ./autogen.sh
+RUN ./configure CC="${CC}" CXX="${CXX}" CFLAGS="${CFLAGS}" CXXFLAGS="${CXXFLAGS}"
+RUN sh -c 'make -j $(nproc)'
+RUN make install
```

Comment:
I was wondering if it'd make sense to put `-j $(nproc)` for install as well.

Reply:
possibly? install is very fast -- doesn't really matter.